app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -82,8 +82,19 @@ def correct_puppet_environment
       end
 
       def content_and_puppet_matched?
-        content_facet && content_facet.content_view_id_was == environment.try(:content_view).try(:id) &&
-          content_facet.lifecycle_environment_id_was == self.environment.try(:lifecycle_environment).try(:id)
+        # IMPORTANT:
+        #
+        # Since rails 5.2, _before_last_save and _was exists on different stages
+        # of callback cycles the _was exists on "before_save", and
+        # "_before_last_save" "after_save" here the following method is called
+        # on both callback hooks, so it is validating both cases.
+        #
+        # Additional important thing to know, is that "_was" changes it's value
+        # after "save" was called, and it is pointing to the current value of
+        # that field, so just using the "_was" will trigger it's own issues.
```

Comment:
This style of comment seems more appropriate as a commit message rather than a comment in the code, do you mind moving it to the commit?

Reply:
Everything else looks great and already covered by unit tests!